docs/_static/source-in-links.js

```diff
@@ -0,0 +1,20 @@
+function addUTMSourceToLinks(href) {
+    try {
+        var url = new URL(href);
+        url.searchParams.set("utm_source", "docarray");
+        return url.href
+    }
+    catch{}
```

Comment:
Maybe just return href in the catch clause ?

Reply:
Catch block is executed only if there's an error in try block. If there are no errors, nothing will be returned from that function

Reply:
no the logic is, attempt to add utm source to link, otherwise if failed (catch block), keep it as it is (return href as it is)